src/PrestaShopBundle/Controller/Admin/Improve/Design/MailThemeController.php

```diff
@@ -0,0 +1,188 @@
+{
+    /**
+     * Show mail theme settings and generation page.
+     *
+     * @AdminSecurity("is_granted('read', request.get('_legacy_controller'))")
+     *
+     * @param Request $request
+     *
+     * @return Response
+     */
+    public function indexAction(Request $request)
+    {
+        $legacyController = $request->attributes->get('_legacy_controller');
+        $generateThemeMailsForm = $this->createForm(GenerateMailsType::class);
+
+        return $this->render('@PrestaShop/Admin/Improve/Design/MailTheme/index.html.twig', [
+            'layoutHeaderToolbarBtn' => [],
+            'layoutTitle' => $this->trans('Mail Theme', 'Admin.Navigation.Menu'),
+            'requireAddonsSearch' => true,
+            'enableSidebar' => true,
+            'help_link' => $this->generateSidebarLink($legacyController),
+            'mailThemeConfigurationForm' => $this->getMailThemeFormHandler()->getForm()->createView(),
+            'generateMailsForm' => $generateThemeMailsForm->createView(),
+        ]);
+    }
+
+    /**
+     * Manage generation form post and generate mails.
+     *
+     * @AdminSecurity("is_granted(['create', 'update'], request.get('_legacy_controller'))")
+     *
+     * @param Request $request
+     *
+     * @return Response
+     */
+    public function generateMailsAction(Request $request)
+    {
+        $generateThemeMailsForm = $this->createForm(GenerateMailsType::class);
+        $generateThemeMailsForm->handleRequest($request);
+
+        if ($generateThemeMailsForm->isSubmitted()) {
+            if (!$generateThemeMailsForm->isValid()) {
+                $this->flashErrors($this->getFormErrorsForJS($generateThemeMailsForm));
+
+                return $this->redirectToRoute('admin_mail_theme_index');
+            }
+
+            $data = $generateThemeMailsForm->getData();
+            try {
+                /** @var GenerateMailTemplatesService $generator */
+                $generator = $this->get('prestashop.service.generate_mail_templates');
+                //Overwrite theme folder if selected
+                if (!empty($data['theme'])) {
+                    $themeFolder = $this->getParameter('themes_dir') . $data['theme'];
+                    $generator
+                        ->setCoreMailsFolder($themeFolder . '/mails')
+                        ->setModulesMailFolder($themeFolder . '/modules')
+                    ;
+                }
+
+                $generator->generateMailTemplates($data['mail_theme'], $data['language'], $data['overwrite']);
+
+                $flashMessage = 'Successfully generated mail templates for theme %s with locale %s';
+                if ($data['overwrite']) {
+                    $flashMessage = 'Successfully overrode mail templates for theme %s with locale %s';
+                }
+                $this->addFlash(
+                    'success',
+                    $this->trans(
+                        sprintf(
+                            $flashMessage,
```

Comment:
Where is the wording here?

Reply:
it's in a variable, I'm gonna change the code to use a string, I'm not sure TranslationTools is able to extract the keys if we use variables